tests/create-testing-library-rule.test.ts

```diff
@@ -89,6 +89,102 @@ ruleTester.run(RULE_NAME, rule, {
       import { foo } from 'custom-module-forced-report'
     `,
     },
+
+    // Test Cases for all settings mixed
```

Comment:
Adding tons of test cases here for new detection helpers. Even if they are covered by rules themselves, I find this really useful to:

1. document its intended usage
2. have specific test for each detection helper in case the real rules are modified
3. implement detection helpers isolated first and then include them within real rules
4. check all necessary conditions are met to report errors (i.e. `canReportErrors`) so each rule doesn't have to repeat these tests (although I've repeated some in few rules, just to give some context within rule tests themselves)